pkg/controller_test.go

```diff
@@ -263,6 +361,24 @@ func fakeController(pairs ...testPod) *controller {
 	}
 }
 
+// Runs the asynchronous goroutines and ensures they complete. Workloads need to be
+// scheduled beforehand.
+func (c *controller) sync() {
```

Comment:
Re-reading my PR: Maybe the func should be called `terminate_graceful`. Then the comment can be removed as the intention should be clear.